2_correlation_analysis/spell_vs_counts_experiment/1_correlation_analysis.py

```diff
@@ -0,0 +1,253 @@
+assert pa14_corr_original[pa14_corr_original.duplicated(keep=False)].shape[0] == 0
+
+print(pao1_corr_original.shape)
+pao1_corr_original.head()
+
+print(pa14_corr_original.shape)
+pa14_corr_original.head()
+
+# +
+# %%time
+# Plot heatmap
+o1 = sns.clustermap(pao1_corr_original, cmap="BrBG", center=0, figsize=(20, 20))
+o1.fig.suptitle("Correlation of raw PAO1 genes", y=1.05, fontsize=24)
+
+# Save
+pao1_pearson_filename = os.path.join(
+    paths.LOCAL_DATA_DIR, f"pao1_{subset_genes}_raw_clustermap_test.png"
+)
+o1.savefig(pao1_pearson_filename, dpi=300)
+
+# +
+# Plot heatmap
+o2 = sns.clustermap(pa14_corr_original, cmap="BrBG", center=0, figsize=(20, 20))
+o2.fig.suptitle("Correlation of raw PA14 genes", y=1.05, fontsize=24)
+
+# Save
+pa14_pearson_filename = os.path.join(
+    paths.LOCAL_DATA_DIR, f"pa14__{subset_genes}_raw_clustermap_test.png"
+)
+o2.savefig(pa14_pearson_filename, dpi=300)
+# -
+
+# ## Log transform + SPELL Correlation
+#
+# _Review of SVD_
+#
+# Singular Value Decomposition is a way to factorize your matrix, $X^{mxn}$ into singular vectors and singular values: $X = U \Sigma V^*$
+#
+# In our case $X$ is **gene x sample** and then the columns of $U$ (gene x eigensample) are the left singular vectors (gene coefficient vectors); $\Sigma$ (eigengene x eigensample) has singular values and is diagonal (mode amplitudes); and $V^T$ (eigengene x sample) has rows that are the right singular vectors (expression level vectors).
+#
+# Here we are using SVD to reduce the noise in our original data by performing dimensionality reduction. This dimensionality is done by neglecting the small singular values in the diagonal matrix $\Sigma$. Normally people would get the projection of the original data onto the singular vectors by $U \Sigma$ and apply the correlation on the projected data. Here, we're following the description in [Hibbs et. al.](https://academic.oup.com/bioinformatics/article/23/20/2692/229926) where they performed correlation on $U$ only.
+#
+# From [Hibbs et. al.](https://academic.oup.com/bioinformatics/article/23/20/2692/229926), we apply their "signal balancing technique that enhances biological information". This is the first part of their [SPELL](https://spell.yeastgenome.org/) algorithm that is described in section 2.3.1. SPELL calculates the correlation on the gene coefficient matrix, $U$ (i.e. how much genes contribute to a latent variable) that is generated after applying SVD. This matrix represents how genes contribute to independent latent variables that capture the signal in the data where the variance of the variables is 1. The idea is that correlations between gene contributions are more balanced so that less prominent patterns are amplified and more dominant patterns are dampended due to this compression. Figure 3 shows how well SPELL recapitulates biology (i.e. the relationship between genes within a GO term) compared to Pearson correlation.
```

Comment:
Oh, now I see how this works. It does sound very similar to using ADAGE to calculate correlation over all the latent variables (except in this case you're using linear "latent variables" or singular vectors). This is a nice description!